contracts/interfaces/IStakingRewardsCampaign.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity 0.8.17;
+
+interface IStakingRewardsCampaign {
+    // enum, structs
+    enum LockType {
+        STAKE_0, 
```

Comment:
```
mapping(LockType => LockInfo) public lockPeriod; 
mapping(address => mapping(uint64 => mapping(LockType => StakeInfo))) private stakeInfo;
```

would become this: (you have to convert the string to a fixed string using bytes, e.g. bytes16)

```
mapping(bytes16 => LockInfo) public lockPeriod;
mapping(address => mapping(bytes16 => StakeInfo)) private stakeInfo;
```

This might be a simpler mapping and allows us to both re-use existing staking, as well as add more at any time.

Ethers or similar libraries should have utility functions to make this conversion, like `utils.formatBytes32String`.

Reply:
As an alternative, campaignType could be a bytes16 instead of a numeric type; but it is less granular and only adds more code, where the other method should actually remove a little bit of code.